service/server.py

```diff
@@ -1,39 +1,138 @@
 import logging
 import logging.config
+import os
+import re
+import requests
+import time
+from wtforms.fields import StringField, PasswordField
+from wtforms.validators import Required, Length
+
+from service import app, login_manager
+
+
+REGISTER_TITLE_API = app.config['REGISTER_TITLE_API']
+LOGIN_JSON = '{{"credentials":{{"user_id":"{}","password":"{}"}}}}'
+FORWARD_SLASH = '/'
+UNAUTHORISED_WORDING = 'There was an error with your Username/Password combination. Please try again'
+GOOGLE_ANALYTICS_API_KEY = app.config['GOOGLE_ANALYTICS_API_KEY']
+LOGGER = logging.getLogger(__name__)
+
+
+class User():
+    def __init__(self, username):
+        self.user_id = username
+
+    def get_id(self):
+        return self.user_id
+
+    def is_authenticated(self):
+        return True
 
-logger = logging.getLogger(__name__)
-register_title_api = app.config['REGISTER_TITLE_API']
-google_analytics_api_key = app.config['GOOGLE_ANALYTICS_API_KEY']
+    def is_active(self):
+        return True
+
+    def is_anonymous(self):
+        return False
+
+
+class LoginApiClient():
+    def __init__(self, login_api_url):
+        self.authentication_endpoint_url = '{}user/authenticate'.format(login_api_url)
+    
+    def authenticate_user(self, username, password):
+        formatted_json = LOGIN_JSON.format(username, password)
+        headers = {'content-type': 'application/json'}
+        response = requests.post(self.authentication_endpoint_url, data=formatted_json, headers=headers)
+        return response.status_code == 200
+
+
+LOGIN_API_CLIENT = LoginApiClient(app.config['LOGIN_API'])
+
+
+@login_manager.user_loader
+def load_user(user_id):
+    return User(user_id)
 
-# TODO Create a proper secret key and store it securely
-app.secret_key = 'a_secret_key'
 
 @app.route('/', methods=['GET'])
 def home():
-    return render_template('home.html', asset_path = '../static/')
+    return render_template('home.html', asset_path='../static/')
+
+
+@app.route('/login', methods=['GET'])
+def signin_page():
+    return render_template(
+        'display_login.html',
+        asset_path='../static/',
+        form=SigninForm(csrf_enabled=_is_csrf_enabled())
+    )
+
+
+BAD_LOGIN_COUNTER = Counter()
+MAX_LOGIN_ATTEMPTS = 10
+NOF_SECS_BETWEEN_LOGINS = 2
+
+
+@app.route('/login', methods=['POST'])
+def signin():
+    form = SigninForm(csrf_enabled=_is_csrf_enabled())
+    if not form.validate():
+        # entered details from login form incorrectly so send back to same page with form error messages
+        return render_template('display_login.html', asset_path='../static/', form=form)
+
+    next_url = request.args.get('next', 'title-search')
+
+    # form was valid
+    username = form.username.data
+    too_many_bad_logins = BAD_LOGIN_COUNTER[username] > MAX_LOGIN_ATTEMPTS
+    if not too_many_bad_logins:
+        # form has correct details. Now need to check authorisation
+        authorised = LOGIN_API_CLIENT.authenticate_user(username, form.password.data)
```

Comment:
we seem to be passing through parameters to the API exactly as the user enters them here, what would happen if I enter a json-like structure in the fields that would make the JSON being sent to the login-api invalid?